### PR TITLE
chore: upgrade material ui to v5

### PR DESCRIPTION
**Pull request by sandrica89**
- Allow yarn >= 1.19
- Upgrade to node 14
- Upgrade to react 17
- Upgrade material ui to v5:  https://mui.com/guides/migration-v4/
- Upgrade dependencies
- Apply new test conventions (cf mattermost for more info)

Troubleshooting that may be discussed:
- error when running tests with apollo-link-timeout (solution applied: mock the dependency in all tests)

Trello cards:
- https://trello.com/c/Ga2bVqJ8/502-upgrade-node-yarn-react-front
- https://trello.com/c/JTguJ5lZ/453-migrer-la-version-de-material-ui

### CODE REVIEW

File: web/admin/Admin.js [hunk 1]
```diff
@@ -46,7 +47,29 @@ const useStyles = makeStyles(theme => ({
...
+      const matches = useMediaQuery(theme.breakpoints.up(key));
+      return !output && matches ? key : output;
+    }, null) || "xs"
+  );
+}
+
+function useIsWidthUp(breakpoint) {
+  const theme = useTheme();
+  return useMediaQuery(theme.breakpoints.up(breakpoint));
+}
```
Comment (sandrica89) [inline web/admin/Admin.js:70]: These 2 hooks should be extracted. They are duplicated among the code.

Comment (sandrica89) [inline web/admin/Admin.js:70]: These 2 hooks should be extracted. They are duplicated among the code.

**Comment by rtaieb** [comment]
There is also a little problem when I want to modify an activity of my current mission : when I click on the start date, I can only change the day, and not the hour

**Comment by rtaieb** [comment]
When I use Geolocation in the "Quel est le lieu de prise de service ?" field, if I type for example "Jussieu", then I replace it with "République", the addresses that match "Jussieu" will remain alongside the ones that match République

**Review by rtaieb — APPROVED** [review]
great job :1st_place_medal:

**Pull request merged**